### PR TITLE
add missing Python dependency to PETSc 3.12.x easyconfigs

### PR DESCRIPTION
**Pull request by lexming**
(created using `eb --new-pr`)

Fix for issue #10865 

Depends on ~~https://github.com/easybuilders/easybuild-easyblocks/pull/2101~~

### CODE REVIEW

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 11 out of 11 (5 easyconfigs in this PR)
node374.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/74f3984e6c4d97f8887782b28d320b7b for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 11 out of 11 (5 easyconfigs in this PR)
node117.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/0d00177183987a543af82ccfb54bdf92 for a full test report.

File: easybuild/easyconfigs/p/PETSc/PETSc-3.12.3-foss-2019b.eb [hunk 1]
```diff
@@ -19,6 +19,8 @@ checksums = [
     '64cf9d5008d5e92117e65bdec5316d991b6a6b8c8ecf7ea46eb790a498266297',  # PETSc_ranlib-fix.patch
 ]
 
+osdependencies = [('numpy', 'python-numpy')]
```
Comment (boegel) [inline easybuild/easyconfigs/p/PETSc/PETSc-3.12.3-foss-2019b.eb:22]: I'm really not in favor of this...

The proper way forward is to replace this `PETSc` easyconfig with a new one that depend on `Python` (and has the corresponding `versionsuffix`), and fixing any easyconfigs that depend on it as well.

Not doing so was a mistake, so we should fix it properly, not sweep it under the rug...

Comment (lexming) [reply easybuild/easyconfigs/p/PETSc/PETSc-3.12.3-foss-2019b.eb:22]: My intention was to ensure that current installations of `PETSc` in production can be reproduced in the future. Completely replacing the existing easyconfigs is also an option of course. In such a case we can rely on older EB releases to reproduce current installations.

Comment (boegel) [reply easybuild/easyconfigs/p/PETSc/PETSc-3.12.3-foss-2019b.eb:22]: Indeed, I think that's the best way forward...

Sometimes bug fixes warrant "breaking" changes, this is a case like that imho...

**Comment by boegel** [comment]
@lexming You'll need to tweak the `PETSc` dependency in `gmsh-4.5.6-foss-2019b-Python-3.7.4.eb` and `SLEPc-3.12.2-foss-2019b.eb` (and also add the missing `versionsuffix` to the latter)?

**Comment by lexming** [comment]
@boegel yep, I'll come back to this issue asap

**Comment by lexming** [comment]
@boegel this is ready for review

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in this PR)
node377.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/db8c86c0d90c5c7b1bc8224c25dc14f8 for a full test report.

**Comment by boegel** [comment]
Tests failed in GitHub Actions, see https://github.com/easybuilders/easybuild-easyconfigs/actions/runs/170140642

**Comment by boegel** [comment]
@boegelbot: please test

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@lexming Something is off with `SLEPc`?

```
== 2020-07-19 18:41:02,687 run.py:222 INFO running cmd:  ./configure --prefix=/users/boegelbot/CentOS8/haswell/software/SLEPc/3.12.2-foss-2019b-Python-3.7.4  --with-python=1 --with-python-dir=/shared/easybuilder/CentOS8/haswell/software/Python/3.7.4-GCCcore-8.3.0  
== 2020-07-19 18:41:02,893 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): cmd " ./configure --prefix=/users/boegelbot/CentOS8/haswell/software/SLEPc/3.12.2-foss-2019b-Python-3.7.4  --with-python=1 --with-python-dir=/shared/easybuilder/CentOS8/haswell/software/Python/3.7.4-GCCcore-8.3.0 " exited with exit code 1 and output:
ERROR: Invalid arguments --with-python=1 --with-python-dir=/shared/easybuilder/CentOS8/haswell/software/Python/3.7.4-GCCcore-8.3.0
Use -h for help
```

**Comment by lexming** [comment]
@boegel this PR depends on https://github.com/easybuilders/easybuild-easyblocks/pull/2101

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 2101"

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in this PR)
node3401.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/bf5a3d456c53f2bb2bfc4ef7d70a2dba for a full test report.

**Comment by boegel** [comment]
@lexming `ELPA` installation failed on CentOS 8 with:

```
/usr/bin/env: python: No such file or directory
```

There's no `python` command in that CentOS 8 setup (only `python2` and `python3`), but it's an easy fix: just add `Python` as a build dep for `ELPA`. I'll open a separate PR for that.

**Comment by lexming** [comment]
@boegel I wonder why CentOS devs decided to not alias `python` to `python3` if there is no other Python installation shipped by default. That's going to break many things... In any case, making a new PR for `ELPA` seems fine.

**Comment by boegel** [comment]
> @boegel I wonder why CentOS devs decided to not alias `python` to `python3` if there is no other Python installation shipped by default. That's going to break many things... In any case, making a new PR for `ELPA` seems fine.

You can alias `python`, but it's opt-in, and it's up to you whether you alias it to `python2` or `python3`.

It gets more interesting: there's also a 3rd Python installation, the "platform Python", which "you should never use".
See https://developers.redhat.com/blog/2018/11/14/python-in-rhel-8/.

**Comment by lexming** [comment]
@boegel very interesting read, thanks for the link!

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 2101"

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 10 out of 10 (10 easyconfigs in this PR)
node2434.golett.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/13c197fd1f3c16ea4fdf261a6d1f1195 for a full test report.

**Pull request merged**